File: backend/src/main/java/com/back/domain/pet/dto/response/PetInfoResponseDto.java
package com.back.domain.pet.dto.response;

import com.back.domain.pet.entity.Pet;
import java.time.LocalDateTime;
import lombok.Builder;

import java.util.List;
import java.util.stream.Collectors;

@Builder
public record PetInfoResponseDto(
        Long id,
        Long petOwnerId,
        String name,
        String species,
        int age,
        String gender,
        String description,
        String imageUrl,
        String shelterName,
        LocalDateTime createdAt,
        List<String> petStatuses
) {
    public static PetInfoResponseDto from(Pet pet) {
        return PetInfoResponseDto.builder()
                .id(pet.getId())
                .petOwnerId(pet.getMember().getId())
                .name(pet.getName())
                .species(pet.getSpecies())
                .age(pet.getAge())
                .gender(pet.getGender().name())
                .description(pet.getDescription())
                .imageUrl(pet.getImageUrl())
                .shelterName(getShelterName(pet))
                .createdAt(pet.getCreatedAt())
                .petStatuses(
                        pet.getPetStatuses().stream()
                                .map(ps -> ps.getStatus().name())
                                .collect(Collectors.toList())
                )
                .build();
    }

    private static String getShelterName(Pet pet) {
        return pet.getShelter() != null ? pet.getShelter().getName() : "보호소 정보 없음";
    }
}

File: backend/src/main/java/com/back/domain/pet/dto/response/PetInfoResponseDto.kt
package com.back.domain.pet.dto.response

import com.back.domain.pet.entity.Pet
import java.time.LocalDateTime

data class PetInfoResponseDto(
    val id: Long,
    val petOwnerId: Long, // 항상 존재
    val name: String,
    val species: String,
    val age: Int,
    val gender: String,
    val description: String?,
    val imageUrl: String?,
    val shelterName: String, // 보호소 정보 없으면 기본값 처리
    val createdAt: LocalDateTime?,
    val petStatuses: List<String>
) {
    companion object {
        fun from(pet: Pet): PetInfoResponseDto {
            return PetInfoResponseDto(
                id = pet.id!!,
                petOwnerId = pet.member!!.id, // 항상 존재하므로 !!
                name = pet.name,
                species = pet.species,
                age = pet.age,
                gender = pet.gender.name,
                description = pet.description,
                imageUrl = pet.imageUrl,
                shelterName = pet.shelter?.name ?: "보호소 정보 없음",
                createdAt = pet.createdAt,
                petStatuses = pet.petStatuses.map { it.status.name }
            )
        }
    }
}